src/transformers/quantizers/quantizer_compressed_tensors.py

```diff
@@ -63,20 +67,59 @@ def _process_model_before_weight_loading(self, model, **kwargs):
         from compressed_tensors.quantization import apply_quantization_config
 
         ct_quantization_config = self.compressor.quantization_config
-        apply_quantization_config(model, ct_quantization_config, run_compressed=True)
 
-    def _process_model_after_weight_loading(self, model, **kwargs) -> None:
-        pass
+        # populate scales and zp
+        if self.run_compressed and self.is_quantization_compressed:
+            apply_quantization_config(model, ct_quantization_config, run_compressed=True)
+        elif not self.is_quantization_compressed:  # Just quantized model or sparse model
+            apply_quantization_config(model, ct_quantization_config)
```

Comment:
wont this fail for sparse only?

Reply:
Good question. 
ct_quantization_config is None for sparse only case, so apply_quantization_config will just return